### PR TITLE
FOGL-5483 - User API fixes

### PR DESCRIPTION
**Pull request by ashish-jabble**
Package based system tests are passing with the change

### CODE REVIEW

File: python/fledge/services/core/api/auth.py [hunk 1]
```diff
@@ -348,39 +349,38 @@ async def create_user(request):
...
+             curl -H "authorization: <token>" -X PUT -d '{"real_name": "AJ"}' http://localhost:8081/fledge/user
     """
     if request.is_auth_optional:
         _logger.warning(FORBIDDEN_MSG)
         raise web.HTTPForbidden
 
-    uid = request.match_info.get('id')
+    uid = request.match_info.get('user_id')
+    if not request.user_is_admin or int(request.user["id"]) != int(uid):
+        msg = "admin privileges are required to logout other user"
```
Comment (praveen-garg) [inline python/fledge/services/core/api/auth.py:367]: logout -> update

File: python/fledge/services/core/api/auth.py [hunk 2]
```diff
@@ -389,6 +389,71 @@ async def update_user(request):
...
+            msg = "Access method should not be empty"
+            raise web.HTTPBadRequest(reason=msg, body=json.dumps({"message": msg}))
+        else:
+            valid_access_method = ('any', 'pwd', 'cert')
+            if access_method not in valid_access_method:
+                msg = "Accepted access method values are {}".format(valid_access_method)
+                raise web.HTTPBadRequest(reason=msg, body=json.dumps({"message": msg}))
+            user_data.update({"access_method": access_method})
+    if 'description' in data:
+        if len(description.strip()) == 0:
```
Comment (praveen-garg) [inline python/fledge/services/core/api/auth.py:433]: can be empty

File: python/fledge/services/core/api/auth.py [hunk 3]
```diff
@@ -35,7 +35,8 @@
     | PUT                        | /fledge/{user_id}/logout                           |
     
     | POST                       | /fledge/admin/user                                 |
-    | PUT                        | /fledge/admin/{user_id}/enabled                    |
+    | PUT                        | /fledge/admin/user/{user_id}                       |
```
Comment (praveen-garg) [inline python/fledge/services/core/api/auth.py:38]: /user/ is redundant here, and not consistent with other admin specific urls

File: python/fledge/services/core/api/auth.py (hunk 1 above)
Comment (ashish-jabble) [reply python/fledge/services/core/api/auth.py]: Even this user based check is not required. As update on the basis of token. Hence removed

Comment (praveen-garg) [reply python/fledge/services/core/api/auth.py]: perfect!

**Review by praveen-garg — APPROVED** [review]

**Review by monika-Dianomic — APPROVED** [review]

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**